joomla/joomla-cms
# [4.0] Fix + Simplify code of frontend module editing

Pull Request for Issue # .

### Summary of Changes
com_config (frontend) has really strange code. Instead of calling backend model getItem() method to get the module data, it uses complicated code by calling controller display method to get json view data.

This PR fixes that strange code. As a result, the json view + some code in com_modules is removed, too.

### Testing Instructions
1. Install Joomla 4 (with sample data for easier testing)
2. Login to frontend using a super admin account
3. Try to edit a module
3. Before patch, no form display to allow edit module parameters
4. After patch, a form displayed to allow changing module parameter. Save the module data, make sure the changes parameter is saved.
